hyperon_das/index.py

```diff
@@ -0,0 +1,72 @@
+from dataclasses import dataclass
+from typing import Optional
+
+
+class QueryOperators:
```

Comment:
DAS shouldn't have any knowledge about the DB backend being used. So this class don't belong to this repository. It should be in das-atomdb. And the class' documentation comments say it's specifically for MongoDB. If that's the case, it shpould be renamed accordingly. Something like "MongoDBQueryOperators" or something alike.

Reply:
@marcocapozzoli 
I know we already had this discussion but if it's not possible (or if it's somehow difficult) to have a generic class for query operators instead of a mongodb-focused one, I think you should reconsider the implementation of the `create_partial_field_index()` method.

Reply:
And "Query Operator" is somehow a misleading name as it's not exactly an operator like in relational algebra operator. They are just logical connectors to write boolean expressions. Perhaps something like "Query Boolean Connectors" or "Query Boolean Operators" are better names for this class.

Reply:
@andre-senna 
I agree with you about where this class should be and about name

What do you think of the name `MongoQuerySelectors`. This is the name that the MongoDB itself uses [Query selectors](https://www.mongodb.com/docs/manual/reference/operator/query/#query-selectors)

The idea was to create the indexes only in MongoDB, if we are going to create it in Redis too, I think this can be done later. So I agree to change the name and make it more specific. but I don't understand your comment about disregarding the implementation of `create_partial_field_index()`? Is it because this method should also create index in Redis?

Reply:
The suggested name is perfect.

I mean, I see no way to implement `create_partial_field_index()` in DAS without exposing a MongoDB-specific class (i.e. MongoQuerySelectors).

Reply:
@andre-senna

I see one way.

The DAS API just passes the request to the QueryEngines classes. So this `LocalQueryEngine` and `RemoteQueryEngine` class knows who their backend is. My idea is to transfer the `MongoQuerySelectors` class to the `atom_db` repository and create a factory function there, and when I pass the class name the function returns the appropriate instance. Example:

`das = DistributedAtomSpace(atom_db='redis_mongo' ...)`

Inside `LocalQueryEngine` I can get the backend class name like this (or similar):

`backend_name = self.local_backend.__class__.__name__`

And I can pass this backend_name to constructor of my Index class

`Index(backend=backend_name, collection=atom_type, key=field, **kwargs).create()`

Inside the constructor I can do something like this:

```
def __init__(...):
    ...
    new_kwargs['conditionals'] = select_selectors(backend).EXISTS(**kwargs)
    ...
```

Where this `select selectors()` function I can create there in atom_db. Something like this:

```
def select_selectors(class_name: str):
    if class_name == "RedisMongoDB":
        return MongoQuerySelectors()
    elif class_name == "OtherDB":
        return OtherDBQuerySelectors()
    ...
```
    
Therefore, this way I don't expose anything about the specific MongoDB class in the DAS and the `select_selectors(...)` function is responsible for this, providing the specific class based on the `atom_db` parameter when the `DistributedAtomSpace(atom_db=...)` was created.

What do you think?

**Ps: the names `select_selectors` and others would not be those. It is an example**